mcs/class/System.IO.Compression/ZipArchive.cs

```diff
@@ -227,13 +227,20 @@ public ZipArchiveEntry GetEntry (string entryName)
 
 		private void Save()
 		{
-			using (var newZip = new MemoryStream()) {
-				zipFile.SaveTo(newZip, CompressionType.Deflate, entryNameEncoding ?? Encoding.UTF8);
+			if (mode == ZipArchiveMode.Create)
+			{
+				zipFile.SaveTo(stream, CompressionType.Deflate, entryNameEncoding ?? Encoding.UTF8);
```

Comment:
If i understand correctly, then the reason for saving to a MemoryStream first, is that the underlying zipFile might try to read from the stream in update mode. But in Create mode we are not reading from the output stream.